enclave/core/sgx/exception.c

```diff
@@ -279,6 +279,9 @@ void oe_real_exception_dispatcher(oe_context_t* oe_context)
         td->host_rbp = td->host_previous_rbp;
         td->host_rsp = td->host_previous_rsp;
 
+        if (oe_exception_record.continuation_callback)
```

Comment:
I'm unclear on why the callback is necessary:
* This seems to be driven by the need to reset the `td->host_rbp`|`rsp` values so that the ocall doesn't end up in the wrong place on the host stack.
* Both handled and unhandled exceptions need to restore these values anyway. Why aren't they simply restored on entry into this method, in which case the exception handler implementation itself can ocall without needing to to defer that to a continuation callback.
* If there is some behavior that cannot be performed in exception handler after those values are restored, how do you message/prevent a continuation callback from trying to do these things and falling over?

Reply:
@mikbras mentioned that there are legitimate cases where one would want their exception handler before these registers are restored, although no concrete examples come to mind for me.

One thing that hangs me up from just doing as you suggest is that other exception handling mechanisms (Windows Vectored Exception Handling, POSIX signals) also call the exception handler before the execution state is restored. But I don't have a good idea of why.

I'll leave this PR open but I think I need to do some more research about why these other mechanisms behave the way they do. I would love if adding this extra handler weren't necessary.

Reply:
@jhand2 Thanks for looking into this. If it turns out that this is necessary, then we should elaborate on the distinction between the two cases in documents and consider adding a usage sample to illustrate what to do in one versus the other. e.g. there's no explanation of the difference of the state of the registers in our public docs today.